lib/azure/core/http/http_request.rb

```diff
@@ -159,7 +159,10 @@ def apply_body_headers
         end
 
         def apply_io_headers
-          headers['Content-Length'] = body.size.to_s
+          headers['Content-Length'] = body.size.to_s if body.respond_to?('size')
+          if headers['Content-Length'].nil?
+            raise ArgumentError, '\'Content-Length\' must be defined if size cannot be get from body IO.'
```

Comment:
should we say "cannot be *obtained* from body IO"?

Reply:
Sure, that's definitely better, will modify the PR soon.

Reply:
looks good, thanks!